go-controller/pkg/clustermanager/secondary_network_cluster_controller.go

```diff
@@ -0,0 +1,321 @@
+}
+
+// RecordUpdateEvent records the udpate event on this given object.
+func (h *secondaryNetworkClusterControllerEventHandler) RecordUpdateEvent(obj interface{}) {
+}
+
+// RecordDeleteEvent records the delete event on this given object.
+func (h *secondaryNetworkClusterControllerEventHandler) RecordDeleteEvent(obj interface{}) {
+}
+
+// RecordSuccessEvent records the success event on this given object.
+func (h *secondaryNetworkClusterControllerEventHandler) RecordSuccessEvent(obj interface{}) {
+}
+
+// RecordErrorEvent records the error event on this given object.
+func (h *secondaryNetworkClusterControllerEventHandler) RecordErrorEvent(obj interface{}, reason string, err error) {
+}
+
+func (h *secondaryNetworkClusterControllerEventHandler) IsResourceScheduled(obj interface{}) bool {
+	return true
+}
+
+// IsObjectInTerminalState returns true if the object is in a terminal state.
+func (h *secondaryNetworkClusterControllerEventHandler) IsObjectInTerminalState(bj interface{}) bool {
+	return false
+}
+
+// AddResource adds the specified object to the cluster according to its type and returns the error,
+// if any, yielded during object creation.
+// Given an object to add and a boolean specifying if the function was executed from iterateRetryResources
+func (h *secondaryNetworkClusterControllerEventHandler) AddResource(obj interface{}, fromRetryLoop bool) error {
+	switch h.objType {
+	case factory.NodeType:
+		node, ok := obj.(*kapi.Node)
+		if !ok {
+			return fmt.Errorf("could not cast %T object to *kapi.Node", obj)
+		}
+		var nodeParams *nodeSyncs
+		if fromRetryLoop {
+			_, nodeSync := h.sncc.addNodeFailed.Load(node.Name)
+			nodeParams = &nodeSyncs{syncNode: nodeSync}
+		} else {
+			nodeParams = &nodeSyncs{syncNode: true}
+		}
+
+		if err := h.sncc.addUpdateNodeEvent(node, nodeParams); err != nil {
+			klog.Errorf("Node add failed for %s, will try again later: %v",
+				node.Name, err)
+			return err
+		}
+	}
+	return nil
+}
+
+// UpdateResource updates the specified object in the cluster to its version in newObj according to its
+// type and returns the error, if any, yielded during the object update.
+// Given an old and a new object; The inRetryCache boolean argument is to indicate if the given resource
+// is in the retryCache or not.
+func (h *secondaryNetworkClusterControllerEventHandler) UpdateResource(oldObj, newObj interface{}, inRetryCache bool) error {
+	switch h.objType {
+	case factory.NodeType:
+		newNode, ok := newObj.(*kapi.Node)
+		if !ok {
+			return fmt.Errorf("could not cast newObj of type %T to *kapi.Node", newObj)
+		}
+		_, ok = oldObj.(*kapi.Node)
+		if !ok {
+			return fmt.Errorf("could not cast oldObj of type %T to *kapi.Node", oldObj)
+		}
+		// determine what actually changed in this update
+		_, nodeSync := h.sncc.addNodeFailed.Load(newNode.Name)
+
+		return h.sncc.addUpdateNodeEvent(newNode, &nodeSyncs{syncNode: nodeSync})
```

Comment:
This func should only be called if a node sync is needed. I think remove the `nodeSyncs` struct and just use a bool to control if we should call `addUpdateNodeEvent` or not.